python/lsst/daf/butler/datastore/record_data.py

```diff
@@ -63,15 +61,17 @@ class SerializedDatastoreRecordData(_BaseModelCompat):
     dataset_ids: list[uuid.UUID]
     """List of dataset IDs"""
 
-    records: Mapping[str, Mapping[str, list[_Record]]]
-    """List of records indexed by record class name and table name."""
+    records: Mapping[str, Mapping[str, Mapping[str, list[_Record]]]]
+    """List of records indexed by record class name, dataset ID (encoded as
+    str, because JSON), and opaque table name.
+    """
```

Comment:
It's too bad we can't use opaque table name to get the record class name instead of storing them both, but I see this is preexisting (and just used in `Quantum` and `QuantumBackedButler`, right?) so maybe this is not the time to try to fix that.

Reply:
The mapping from opaque table name to record class would have to exist somewhere when de-serialization happens. I'm not sure what would be the right place for it.

Reply:
I'm fine with leaving this.  DM-41115 may also change this completely anyway.